File: src/main/java/io/vertigo/shiny/components/data/list/ShinyList.java
package io.vertigo.shiny.components.data.list;

import java.util.List;

import io.vertigo.core.lang.Assertion;
import io.vertigo.shiny.components.ShinyComponent;

public record ShinyList(
		String title,
		List<Object> items, //String || ShinyList
		ShinyListType listType) implements ShinyComponent {

	@Override
	public String type() {
		return "ShinyList";
	}

	public ShinyList {
		Assertion.check()
				.isNotNull(items)
				.isNotNull(listType);
	}
}

File: banshee/src/models/ShinyList.ts
import { ShinyComponent } from '../ShinyComponent';
import { ShinyListType } from './ShinyListType';

export class ShinyList implements ShinyComponent {
  title: string;
  items: (string | ShinyList)[];
  listType: ShinyListType;
  type: string = 'list';

  constructor(title: string, items: (string | ShinyList)[], listType: ShinyListType) {
    this.title = title;
    this.items = items;
    this.listType = listType;
  }
}
